experimental/oak_baremetal_loader/src/lookup.rs

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use hashbrown::HashMap;
+use oak_baremetal_communication_channel::schema;
+use std::fs;
+
+pub fn encode_lookup_data<'a>(
+    mut data: HashMap<Vec<u8>, Vec<u8>>,
+) -> oak_idl::utils::Message<schema::LookupData<'a>> {
+    let mut builder = oak_idl::utils::MessageBuilder::default();
+    let entries: Vec<flatbuffers::WIPOffset<schema::LookupDataEntry>> = data
+        .drain()
+        .map(|(key, value)| {
+            let key = builder.create_vector::<u8>(&key);
+            let value = builder.create_vector::<u8>(&value);
+            schema::LookupDataEntry::create(
+                &mut builder,
+                &schema::LookupDataEntryArgs {
+                    key: Some(key),
+                    value: Some(value),
+                },
+            )
+        })
+        .collect();
+    let items = builder.create_vector(&entries);
+    let message =
+        schema::LookupData::create(&mut builder, &schema::LookupDataArgs { items: Some(items) });
+    builder
+        .finish(message)
+        .expect("errored when creating lookup data update message")
+}
+
+pub fn load_lookup_data(
+    file_path: &std::path::PathBuf,
+) -> anyhow::Result<HashMap<Vec<u8>, Vec<u8>>> {
+    let bytes = fs::read(&file_path)
+        .unwrap_or_else(|_| panic!("Couldn't read Wasm file {}", &file_path.display()));
```

Comment:
```suggestion
        .unwrap_or_else(|_| panic!("Couldn't read lookup data file {}", &file_path.display()));
```
I think! :)

Reply:
Why does not this line return a Result, particularly given that the function returns Result? If the panic is intentional might be worth adding a comment to describe why.

Reply:
Yeah you're right, thanks for catching. And this should indeed return an error instead of throwing.